chia/wallet/puzzles/dao_spend_p2_singleton_v2.clsp

```diff
@@ -119,9 +122,12 @@
             (r parent_amount_list)  ; this is the delegated_puzhash
             (sha256tree 0)  ; most coins destroy themselves
             (calculate_delegated_puzzlehash  ; the last coin creates the conditions
-              (c
-                (list CREATE_COIN P2_SINGLETON_PUZHASH (- (+ total (f (r (f parent_amount_list)))) create_coin_sum) (list (f (r TREASURY_SINGLETON_STRUCT))))
-                CONDITIONS_FOR_THIS_ASSET_TYPE
+              (if (> (- (+ total (f (r (f parent_amount_list)))) create_coin_sum) 0)
```

Comment:
Same comment as before with regards to 0 value change. Make it the wallet's problem saves on chain space.
But if you are going to keep this here then consider making `(- (+ total (f (r (f parent_amount_list)))) create_coin_sum)` a stored value instead of calculating it twice. Both for readability and for optimisation.

Reply:
fixed so it's only calculating once